### PR TITLE
Assignments field

### PR DESCRIPTION
**Pull request by damirpolat**
I added a field in `LearnerClust` class for storing cluster assignments of 'train' data. Since every learner saves cluster assignments in a different way in `model`, the new feature makes it easy to just pull out results of clustering for the data the learner is trained on. 

Thanks, 
Damir

### CODE REVIEW

File: R/LearnerClustCobweb.R [hunk 1]
```diff
@@ -42,7 +42,11 @@ LearnerClustCobweb = R6Class("LearnerClustCobweb",
     .train = function(task) {
       pv = self$param_set$get_values(tags = "train")
       ctrl = do.call(RWeka::Weka_control, pv)
-      invoke(RWeka::Cobweb, x = task$data(), control = ctrl)
+      m = invoke(RWeka::Cobweb, x = task$data(), control = ctrl)
+      self$assignments = m$class_ids + 1L
```
Comment (mllg) [inline R/LearnerClustCobweb.R:46]: self$assignments = unname(m$class_ids + 1L)

File: R/LearnerClustEM.R [hunk 2]
```diff
@@ -53,7 +53,11 @@ LearnerClustEM = R6Class("LearnerClustEM",
       pv = self$param_set$get_values(tags = "train")
       names(pv) = chartr("_", "-", names(pv))
       ctrl = do.call(RWeka::Weka_control, pv)
-      invoke(RWeka::make_Weka_clusterer("weka/clusterers/EM"), x = task$data(), control = ctrl)
+      m = invoke(RWeka::make_Weka_clusterer("weka/clusterers/EM"), x = task$data(), control = ctrl)
+      self$assignments = m$class_ids + 1L
```
Comment (mllg) [inline R/LearnerClustEM.R:57]: self$assignments = uname(m$class_ids + 1L)

File: R/LearnerClustKKMeans.R [hunk 3]
```diff
@@ -79,7 +79,9 @@ LearnerClustKKMeans = R6Class("LearnerClustKKMeans",
       check_centers_param(self$param_set$values$centers, task, test_data_frame, "centers")
 
       pv = self$param_set$get_values(tags = "train")
-      invoke(kernlab::kkmeans, x = as.matrix(task$data()), .args = pv)
+      m = invoke(kernlab::kkmeans, x = as.matrix(task$data()), .args = pv)
+      self$assignments = m[seq_len(length(m))]
```
Comment (mllg) [inline R/LearnerClustKKMeans.R:83]: `seq_len(length(m))` -> `seq_along(m)`

File: R/LearnerClustMiniBatchKMeans.R [hunk 4]
```diff
@@ -69,7 +69,15 @@ LearnerClustMiniBatchKMeans = R6Class("LearnerClustMiniBatchKMeans",
       }
 
       pv = self$param_set$get_values(tags = "train")
-      invoke(ClusterR::MiniBatchKmeans, data = task$data(), .args = pv)
+      m = invoke(ClusterR::MiniBatchKmeans, data = task$data(), .args = pv)
+
+      self$assignments = unclass(ClusterR::predict_MBatchKMeans(
+        data = task$data(),
```
Comment (mllg) [inline R/LearnerClustMiniBatchKMeans.R:75]: This looks like a heavy computation is required. Is there a way to turn it off if we don't need the assignments?

File: R/LearnerClustSimpleKMeans.R [hunk 5]
```diff
@@ -57,7 +57,12 @@ LearnerClustSimpleKMeans = R6Class("LearnerClustSimpleKMeans",
       pv = self$param_set$get_values(tags = "train")
       names(pv) = chartr("_", "-", names(pv))
       ctrl = do.call(RWeka::Weka_control, pv)
-      invoke(RWeka::SimpleKMeans, x = task$data(), control = ctrl)
+      m = invoke(RWeka::SimpleKMeans, x = task$data(), control = ctrl)
+
+      self$assignments = m$class_ids + 1L
+      names(self$assignments) = NULL
```
Comment (mllg) [inline R/LearnerClustSimpleKMeans.R:63]: see above.

File: R/LearnerClustXMeans.R [hunk 6]
```diff
@@ -56,7 +56,12 @@ LearnerClustXMeans = R6Class("LearnerClustXMeans",
       pv = self$param_set$get_values(tags = "train")
       names(pv) = chartr("_", "-", names(pv))
       ctrl = do.call(RWeka::Weka_control, pv)
-      invoke(RWeka::XMeans, x = task$data(), control = ctrl)
+      m = invoke(RWeka::XMeans, x = task$data(), control = ctrl)
+
+      self$assignments = m$class_ids + 1L
+      names(self$assignments) = NULL
```
Comment (mllg) [inline R/LearnerClustXMeans.R:62]: see above.

File: R/LearnerClust.R [hunk 7]
```diff
@@ -32,6 +32,10 @@
 LearnerClust = R6Class("LearnerClust",
   inherit = Learner,
   public = list(
+    #' @field assignments (`NULL` | `integer(1)`)\cr
```
Comment (mllg) [inline R/LearnerClust.R:35]: I guess `integer(1)` is wrong; this is a vector, right?

Comment (mllg) [reply R/LearnerClust.R:35]: Also, we should have an autotest to check this field

**Review by mllg — CHANGES_REQUESTED** [review]

File: R/LearnerClustCobweb.R (hunk 1 above)
Comment (damirpolat) [reply R/LearnerClustCobweb.R]: All fixed. Thanks!

File: R/LearnerClustKKMeans.R (hunk 3 above)
Comment (damirpolat) [reply R/LearnerClustKKMeans.R]: I edited that part, thanks!

File: R/LearnerClustMiniBatchKMeans.R (hunk 4 above)
Comment (damirpolat) [reply R/LearnerClustMiniBatchKMeans.R]: I added `save_assignments` field in `LearnerClust` that lets you turn off saving. It's on by default.

File: R/LearnerClust.R (hunk 7 above)
Comment (damirpolat) [reply R/LearnerClust.R]: > I guess `integer(1)` is wrong; this is a vector, right?

Yep, I fixed that.

Comment (damirpolat) [reply R/LearnerClust.R]: > Also, we should have an autotest to check this field

I added a unit test that checks if `assignments` field exists when saving is on, and vice versa.

**Review by mllg — APPROVED** [review]

**Pull request merged**